### PR TITLE
Add edit endpoint on Category

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/EventService.Validation/Category/EditCategoryRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using LT.DigitalOffice.EventService.Models.Dto.Enums;
+using LT.DigitalOffice.EventService.Models.Dto.Requests.Category;
+using LT.DigitalOffice.EventService.Validation.Category.Interfaces;
+using LT.DigitalOffice.EventService.Validation.EventUser.Resources;
+using LT.DigitalOffice.Kernel.Validators;
+using MassTransit.Contracts.Conductor;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace LT.DigitalOffice.EventService.Validation.Category
```
Comment (AnnPostnikova) [inline src/EventService.Validation/Category/EditCategoryRequestValidator.cs]: add ;

File: src/EventService.Validation/Category/EditCategoryRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        nameof(EditCategoryRequest.Color),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => Enum.TryParse(x.value?.ToString(), out CategoryColor _), "Incorrect Color value." }
+        });
+
+      #endregion
+
+      #region IsActive
```
Comment (AnnPostnikova) [inline src/EventService.Validation/Category/EditCategoryRequestValidator.cs]: Name

File: src/EventService.Data/CategoryRepository.cs [hunk 3]
```diff
@@ -38,17 +42,22 @@ public class CategoryRepository : ICategoryRepository
...
+    IDataProvider provider,
+    IHttpContextAccessor httpContextAccessor)
   {
-    _provider = provider;
+    _provider = provider;
+    _httpContextAccessor = httpContextAccessor;
   }
 
-  public bool DoesExistAllAsync(List<Guid> categoriesIds)
+  public async Task<bool> DoesExistAllAsync(List<Guid> categoriesIds)
```
Comment (AnnPostnikova) [inline src/EventService.Data/CategoryRepository.cs]: DoExistAllAsync

**Review by AnnPostnikova — APPROVED** [review]

File: src/EventService.Validation/Event/CreateEventRequestValidator.cs [hunk 4]
```diff
@@ -63,7 +63,7 @@ await userService.CheckUsersExistenceAsync(users.Select(userRequest => userReque
     When(ev => !ev.CategoriesIds.IsNullOrEmpty(), () =>
     {
       RuleFor(ev => ev.CategoriesIds)
-        .Must(categoryRepository.DoesExistAllAsync)
+        .MustAsync(async (categories, _) => await categoryRepository.DoExistAllAsync(categories))
```
Comment (SolovyovDmitry) [inline src/EventService.Validation/Event/CreateEventRequestValidator.cs]: async/await not required

**Review by SolovyovDmitry — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**